broker/setup_broker/setup_broker.sh

```diff
@@ -36,34 +36,80 @@ if [ "$continue_with_setup" != "y" ]; then
+        else
+            testid_suffix="-${testid}"
+        fi
+    fi
+
+    echo "${base_name}${testid_suffix}"
+}
+
 #--- GCP resources used directly in this script
-broker_bucket="${PROJECT_ID}-${survey}-broker_files"
-bq_dataset="${survey}"
-# use test resources, if requested
-# (there must be a better way to do this)
-if [ "$testid" != "False" ]; then
-    broker_bucket="${broker_bucket}-${testid}"
-    bq_dataset="${bq_dataset}_${testid}"
-fi
+broker_bucket=$(define_GCP_resources "${PROJECT_ID}-${survey}-broker_files")
+bq_dataset=$(define_GCP_resources "${survey}")
+# topics and subscriptions involved in writing alert data to BigQuery
+topic_bigquery_import=$(define_GCP_resources "${survey}-bigquery-import")
+subscription_bigquery_import="${topic_bigquery_import}" # BigQuery subscription
+deadletter_topic_bigquery_import=$(define_GCP_resources "${survey}-bigquery-import-deadletter")
+deadletter_subscription_bigquery_import="${deadletter_topic_bigquery_import}"
+
+alerts_table="alerts_${versiontag}"
+
+# function used to create (or delete) GCP resources
+manage_resources() {
+    local mode="$1"  # setup or teardown
+    local environment_type="production"
+
+    if [ "$testid" != "False" ]; then
+        environment_type="testing"
+    fi
 
+    if [ "$mode" = "setup" ]; then
+        # setup resources
+        python3 setup_gcp.py --survey="$survey" --testid="$testid" --confirmed --region="${region}" --versiontag="${versiontag}"
+        gcloud pubsub topics create "${topic_bigquery_import}"
+        gcloud pubsub topics create "${deadletter_topic_bigquery_import}"
+        gcloud pubsub subscriptions create "${deadletter_subscription_bigquery_import}" --topic="${deadletter_topic_bigquery_import}"
+        # in order to create BigQuery subscriptions, ensure that the following service account:
+        # service-<project number>@gcp-sa-pubsub.iam.gserviceaccount.com" has the
+        # bigquery.dataEditor role for each table
+        gcloud pubsub subscriptions create "${subscription_bigquery_import}" \
+            --topic="${topic_bigquery_import}" \
+            --bigquery-table="${PROJECT_ID}:${bq_dataset}.${alerts_table}" \
+            --use-table-schema \
+            --drop-unknown-fields \
+            --dead-letter-topic="${deadletter_topic_bigquery_import}" \
+            --max-delivery-attempts=5 \
+            --dead-letter-topic-project="${PROJECT_ID}"
+    else
+        if [ "$environment_type" = "testing" ]; then
+            # delete testing resources
+            python3 setup_gcp.py --survey="$survey" --testid="$testid" --teardown --confirmed --versiontag="${versiontag}"
+            gcloud pubsub topics delete "${topic_bigquery_import}"
+            gcloud pubsub topics delete "${deadletter_topic_bigquery_import}"
+            gcloud pubsub subscriptions delete "${deadletter_subscription_bigquery_import}"
+            gcloud pubsub subscriptions delete "${subscription_bigquery_import}"
+        fi
```

Comment:
In case the user tries to delete production resources with this script, let's give them some info about why nothing will happen.

```suggestion
        if [ "$environment_type" = "testing" ]; then
            # delete testing resources
            python3 setup_gcp.py --survey="$survey" --testid="$testid" --teardown --confirmed --versiontag="${versiontag}"
            gcloud pubsub topics delete "${topic_bigquery_import}"
            gcloud pubsub topics delete "${deadletter_topic_bigquery_import}"
            gcloud pubsub subscriptions delete "${deadletter_subscription_bigquery_import}"
            gcloud pubsub subscriptions delete "${subscription_bigquery_import}"
        else
            echo 'ERROR: No testid supplied.'
            echo 'To avoid accidents, this script will not delete production resources.'
            echo 'If that is your intention, you must delete them manually.'
            echo 'Otherwise, please supply a testid.'
            exit 1
        fi
```